### ISSUE TITLE
Build fails on Ubuntu's Linux 5.16.0-051600daily20220118-generic

### ISSUE THREAD

**Issue by solardiz** [lkrg-org/lkrg#144]
```
  CC [M]  /root/src/src/modules/notifiers/p_notifiers.o
  CC [M]  /root/src/src/modules/self-defense/hiding/p_hiding.o
/root/src/src/modules/notifiers/p_notifiers.c: In function 'p_register_notifiers':
/root/src/src/modules/notifiers/p_notifiers.c:122:4: error: implicit declaration of function 'task_handoff_register' [-Werror=implicit-function-declaration]
  122 |    task_handoff_register(&p_taskfree_notifier_nb);
      |    ^~~~~~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:123:4: error: implicit declaration of function 'profile_event_register' [-Werror=implicit-function-declaration]
  123 |    profile_event_register(PROFILE_TASK_EXIT, &p_profile_event_exit_notifier_nb);
      |    ^~~~~~~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:123:27: error: 'PROFILE_TASK_EXIT' undeclared (first use in this function)
  123 |    profile_event_register(PROFILE_TASK_EXIT, &p_profile_event_exit_notifier_nb);
      |                           ^~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:123:27: note: each undeclared identifier is reported only once for each function it appears in
/root/src/src/modules/notifiers/p_notifiers.c:124:27: error: 'PROFILE_MUNMAP' undeclared (first use in this function)
  124 |    profile_event_register(PROFILE_MUNMAP, &p_profile_event_munmap_notifier_nb);
      |                           ^~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c: In function 'p_deregister_notifiers':
/root/src/src/modules/notifiers/p_notifiers.c:257:4: error: implicit declaration of function 'task_handoff_unregister' [-Werror=implicit-function-declaration]
  257 |    task_handoff_unregister(&p_taskfree_notifier_nb);
      |    ^~~~~~~~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:258:4: error: implicit declaration of function 'profile_event_unregister' [-Werror=implicit-function-declaration]
  258 |    profile_event_unregister(PROFILE_TASK_EXIT, &p_profile_event_exit_notifier_nb);
      |    ^~~~~~~~~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:258:29: error: 'PROFILE_TASK_EXIT' undeclared (first use in this function)
  258 |    profile_event_unregister(PROFILE_TASK_EXIT, &p_profile_event_exit_notifier_nb);
      |                             ^~~~~~~~~~~~~~~~~
/root/src/src/modules/notifiers/p_notifiers.c:259:29: error: 'PROFILE_MUNMAP' undeclared (first use in this function)
  259 |    profile_event_unregister(PROFILE_MUNMAP, &p_profile_event_munmap_notifier_nb);
      |                             ^~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:288: /root/src/src/modules/notifiers/p_notifiers.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/root/src/src/modules/self-defense/hiding/p_hiding.c: In function 'p_unhide_itself':
/root/src/src/modules/self-defense/hiding/p_hiding.c:77:38: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   77 |    struct kobj_type  *p_tmp_ktype  = p_tmp_mod->mkobj.kobj.ktype;
      |                                      ^~~~~~~~~
make[1]: *** [Makefile:1846: /root/src] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.16.0-051600daily20220118-generic'
```

**Comment by Adam-pi3** [comment]
Latest kernel completely removed profile_task_exit, profile_munmap and task_free_notifier:
https://github.com/torvalds/linux/commit/2873cd31a20c25b5e763b35e5fb886f0938c6dd5
https://lore.kernel.org/all/20220103213312.9144-1-ebiederm@xmission.com/T/#u

It should be fixed by https://github.com/lkrg-org/lkrg/commit/1b39b4cf9902cb239be08ce30b13c0e312770ed1
I'm closing this issue.

**Comment by solardiz** [comment]
Thank you for addressing this, Adam.

Did the issue really affect 5.16 proper? We just got a report of another issue affecting a build of 5.16.2 at module load time, so I guess 5.16.x are actually unaffected by this build time issue so far (and the commit message saying 5.16+ is thus wrong, but is too late for us to fix now). I guess only latest mainline kernels are affected, and they have 5.16.0 there as a placeholder (not yet updated). I didn't look at those different kernels - these are just guesses - but we need to figure this out to add a proper `CHANGES` entry, and possibly update the `README` and the homepage.
